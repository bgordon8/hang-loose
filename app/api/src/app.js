import express from 'express'

const app = express()

app.get("/", (req, res) => {
    res.status(200).json({
        message: "hello world!!!! thats whats up!!!"
    })
})

if (process.env.NODE_ENV !== "test") {
    app.listen(3000, () => {
        console.log('listening on port 3000...')
    })
}

export default app