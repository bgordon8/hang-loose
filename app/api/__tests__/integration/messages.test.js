import supertest from 'supertest'
import app from '../../src/app'

const req = supertest(app)

describe('routes : messages', () => {
  describe('GET /messages', () => {
    it('returns all messages', async () => {
      const res = await req.get('/messages')

      expect(res.status).toBe(200)
      expect(res.body).toHaveProperty('status')
      expect(res.body.status).toBe('success')
      expect(res.body).toHaveProperty('messages')
      expect(res.body.messages[0]).toHaveProperty('id')
      expect(res.body.messages[0].id).toBe(1)
      expect(res.body.messages[0]).toHaveProperty('channelId')
      expect(res.body.messages[0].channelId).toBe(1)
      expect(res.body.messages[0]).toHaveProperty('authorId')
      expect(res.body.messages[0].authorId).toBe(1)
      expect(res.body.messages[0]).toHaveProperty('content')
      expect(res.body.messages[0].content).toBe('this is the first message!')
      expect(res.body).toMatchSnapshot()
    })
  })

  describe('GET /messages/:id', () => {
    it('returns message by id', async () => {
      const res = await req.get('/messages/1')

      expect(res.status).toBe(200)
      expect(res.body).toHaveProperty('status')
      expect(res.body.status).toBe('success')
      expect(res.body).toHaveProperty('message')
      expect(res.body.message).toHaveProperty('id')
      expect(res.body.message.id).toBe(1)
      expect(res.body.message).toHaveProperty('channelId')
      expect(res.body.message.channelId).toBe(1)
      expect(res.body.message).toHaveProperty('authorId')
      expect(res.body.message.authorId).toBe(1)
      expect(res.body.message).toHaveProperty('content')
      expect(res.body.message.content).toBe('this is the first message!')
      expect(res.body).toMatchSnapshot()
    })
  })

  describe('POST /messages', () => {
    it('creates new message', async () => {
      const res = await req.post('/messages').send({
        channelId: 1,
        authorId: 1,
        content: 'new created message!',
      })

      expect(res.status).toBe(200)
      expect(res.body).toHaveProperty('status')
      expect(res.body.status).toBe('success')
      expect(res.body).toHaveProperty('message')
      expect(res.body.message).toHaveProperty('id')
      expect(res.body.message.id).toBe(3)
      expect(res.body.message).toHaveProperty('channelId')
      expect(res.body.message.channelId).toBe(1)
      expect(res.body.message).toHaveProperty('authorId')
      expect(res.body.message.authorId).toBe(1)
      expect(res.body.message).toHaveProperty('content')
      expect(res.body.message.content).toBe('new created message!')
      expect(res.body).toMatchSnapshot({
        message: {
          created_at: expect.any(String),
          updated_at: expect.any(String),
        },
      })
    })
  })

  describe('PUT /messages/:id', () => {
    it('updates message by id', async () => {
      const res = await req.put('/messages/1').send({
        channelId: 1,
        authorId: 1,
        content: 'updated message',
      })

      expect(res.status).toBe(200)
      expect(res.body).toHaveProperty('status')
      expect(res.body.status).toBe('success')
      expect(res.body).toHaveProperty('message')
      expect(res.body.message).toHaveProperty('id')
      expect(res.body.message.id).toBe(1)
      expect(res.body.message).toHaveProperty('channelId')
      expect(res.body.message.channelId).toBe(1)
      expect(res.body.message).toHaveProperty('authorId')
      expect(res.body.message.authorId).toBe(1)
      expect(res.body.message).toHaveProperty('content')
      expect(res.body.message.content).toBe('updated message')
      expect(res.body).toMatchSnapshot()
    })
  })
})
