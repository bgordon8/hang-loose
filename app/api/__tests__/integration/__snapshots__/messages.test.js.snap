// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes : messages DELETE /messages/:id deletes message by id 1`] = `
Object {
  "message": Object {
    "authorId": 1,
    "channelId": 1,
    "content": "this is the first message!",
    "created_at": "2020-10-12T06:00:00.000Z",
    "id": 1,
    "updated_at": "2020-10-12T06:00:00.000Z",
  },
  "status": "success",
}
`;

exports[`routes : messages GET /messages returns all messages 1`] = `
Object {
  "messages": Array [
    Object {
      "authorId": 1,
      "channelId": 1,
      "content": "this is the first message!",
      "created_at": "2020-10-12T06:00:00.000Z",
      "id": 1,
      "updated_at": "2020-10-12T06:00:00.000Z",
    },
    Object {
      "authorId": 2,
      "channelId": 1,
      "content": "this is the second message!",
      "created_at": "2020-10-12T06:00:00.000Z",
      "id": 2,
      "updated_at": "2020-10-12T06:00:00.000Z",
    },
  ],
  "status": "success",
}
`;

exports[`routes : messages GET /messages/:id returns message by id 1`] = `
Object {
  "message": Object {
    "authorId": 1,
    "channelId": 1,
    "content": "this is the first message!",
    "created_at": "2020-10-12T06:00:00.000Z",
    "id": 1,
    "updated_at": "2020-10-12T06:00:00.000Z",
  },
  "status": "success",
}
`;

exports[`routes : messages POST /messages creates new message 1`] = `
Object {
  "message": Object {
    "authorId": 1,
    "channelId": 1,
    "content": "new created message!",
    "created_at": Any<String>,
    "id": 3,
    "updated_at": Any<String>,
  },
  "status": "success",
}
`;

exports[`routes : messages PUT /messages/:id updates message by id 1`] = `
Object {
  "message": Object {
    "authorId": 1,
    "channelId": 1,
    "content": "updated message",
    "created_at": "2020-10-12T06:00:00.000Z",
    "id": 1,
    "updated_at": "2020-10-12T06:00:00.000Z",
  },
  "status": "success",
}
`;
